test/pylib/manager_client.py

```diff
@@ -47,15 +47,26 @@ def __init__(self, sock_path: str, port: int, use_ssl: bool, auth_provider: Any|
         self.ccluster: Optional[CassandraCluster] = None
         self.cql: Optional[CassandraSession] = None
         # A client for communicating with ScyllaClusterManager (server)
-        self.client = UnixRESTClient(sock_path)
+        self.sock_path = sock_path
+        self.client_for_asyncio_loop = {asyncio.get_running_loop(): UnixRESTClient(sock_path)}
         self.api = ScyllaRESTAPIClient()
         self.metrics = ScyllaMetricsClient()
         self.thread_pool = ThreadPoolExecutor()
 
+    @property
+    def client(self):
+        _client = self.client_for_asyncio_loop.get(asyncio.get_running_loop(), None)
+        if _client is None:
+            _client = UnixRESTClient(self.sock_path)
+            self.client_for_asyncio_loop[asyncio.get_running_loop()] = _client
+        return _client
+
     async def stop(self):
         """Close driver"""
         self.driver_close()
-        await self.client.shutdown()
+        # TODO: good candidate for safe_gather  https://github.com/scylladb/scylladb/pull/17781
+        #  to make sure tha all connections is closed
+        await asyncio.gather(*[client.shutdown() for client in self.client_for_asyncio_loop.values()])
```

Comment:
I wonder if we shouldn't do the `shutdown()` on the loop where the client is running. Something like `loop.create_task(client.shutdown()) for loop, client in self.client_for_asyncio_loop`. Dunno if that's legal in asyncio, but I share @nyh concerns that it's weird we have to create the client on another loop, but don't need to shut it down on that loop.

(In the same spirit, shouldn't **all** operations we perform using the client, go to the other loop?)

Reply:
here is interesting. because with 
```loop.create_task(client.shutdown()) for loop, client in self.client_for_asyncio_loop```

it will might be case that @nyh mentioned when loop already finished
> Is the lifetime even guaranteed (e.g., is it possible that some other event loop has already exited)?

as I mentioned,  we are jumping between loops.

for example 
```
    await manager_internal.before_test(test_case_name) - current loop A
    yield manager_internal - go to test body with loop B that created in event_loop fixture
    await manager_internal.after_test(test_case_name, not failed) - loop A again
```


In this case to be complete the save we need to create/close a connection for each ```request``` call(some decorator for example), that will be slow. Or remove event_loop fixture and set manager_internal scope function it will be 1 event_loop and 1 ManagerClient per test  - this also will be slower a bit

Reply:
examples 
```
async def await_test():
  print(123)

loop = asyncio.new_event_loop()
Thread(target=loop.run_forever, daemon=True, name="asyncio_thread").start()
print(loop.is_running())

loop.stop()
asyncio.run_coroutine_threadsafe(await_test(), loop=loop)
time.sleep(5)

```
```
True
Task was destroyed but it is pending!
task: <Task pending name='Task-1' coro=<await_test() running at /home/artsiom/codebase/python-driver/123.py:16> cb=[_chain_future.<locals>._call_set_state() at /home/artsiom/.pyenv/versions/3.11.7/lib/python3.11/asyncio/futures.py:394]>
sys:1: RuntimeWarning: coroutine 'await_test' was never awaited
```


or with close instead of stop
```
True
Task was destroyed but it is pending!
task: <Task pending name='Task-1' coro=<await_test() running at /home/artsiom/codebase/python-driver/123.py:16> cb=[_chain_future.<locals>._call_set_state() at /home/artsiom/.pyenv/versions/3.11.7/lib/python3.11/asyncio/futures.py:394]>
sys:1: RuntimeWarning: coroutine 'await_test' was never awaited

```